crates/ra_syntax/src/grammar/expressions/atom.rs

```diff
@@ -362,7 +366,7 @@ fn match_arm(p: &mut Parser) -> BlockLike {
         patterns::pattern(p);
     }
     if p.eat(IF_KW) {
-        expr_no_struct(p);
```

Comment:
Are struct literals really allowed here? Let's add a test if this is the case?

Reply:
```rust
struct Test{field: u32}

fn main() {
    match Ok(1u32) as Result<u32, u32> {
        Ok(0) if Test{field:0}.field ==0 => Test{field:0},
        _ => Test{field:1}
    };
}
```
Yep - allowed. The only reason they aren't in other places is due to ambiguity, but the `=>` stops  that being an issue anyway.